system/lib/sbrk.c

```diff
@@ -41,8 +53,13 @@ void *sbrk(intptr_t increment) {
 #else
     intptr_t old_brk = *sbrk_ptr;
 #endif
-    // TODO: overflow checks
     intptr_t new_brk = old_brk + increment;
+    // ArrayBuffers are currently limited in practice to 2GB, the size of a
+    // signed integer. When 4GB support arrives, we'll want to add an option
+    // to ignore this check there.
+    if ((uint32_t)new_brk > (uint32_t)INT_MAX) {
+      RETURN_ERROR();
+    }
```

Comment:
Why it this change needed?  Wouldn't `__builtin_wasm_memory_grow` just fail in this case?

Reply:
It seems like it should, yeah, but what I think happens is that wasm VMs use 32-bit signed values for the size, and there's an overflow, so somehow it fails to report the error. Fixing that is part of what supporting 4GB in wasm VMs means, which hopefully will happen.

Reply:
Note that this is a very long-standing issue - we had checks like this in the asm.js sbrk since forever. This new sbrk in C which was added recently just didn't have that check added yet, basically.

Reply:
Can you mention that this is workaround for a VM bug.  It wasn't clear from the comment why it was needed.